src/domain/service/note.ts

```diff
@@ -10,30 +11,50 @@ export default class NoteService {
   /**
    * Note repository
    */
-  public repository: NoteRepository;
+  public noteRepository: NoteRepository;
+
+  /**
+   * Note relationship repository
+   */
+  public noteRelationsRepository: NoteRelationsRepository;
 
   /**
    * Note service constructor
    *
-   * @param repository - note repository
+   * @param noteRepository - note repository
+   * @param noteRelationsRepository - note relationship repository
    */
-  constructor(repository: NoteRepository) {
-    this.repository = repository;
+  constructor(noteRepository: NoteRepository, noteRelationsRepository: NoteRelationsRepository) {
+    this.noteRepository = noteRepository;
+    this.noteRelationsRepository = noteRelationsRepository;
   }
 
   /**
    * Adds note
    *
    * @param content - note content
    * @param creatorId - note creator
+   * @param parentPublicId - parent note if exist
    * @returns { Note } added note object
    */
-  public async addNote(content: JSON, creatorId: Note['creatorId']): Promise<Note> {
-    return await this.repository.addNote({
+  public async addNote(content: JSON, creatorId: Note['creatorId'], parentPublicId: Note['publicId'] | undefined): Promise<Note> {
+    const note = await this.noteRepository.addNote({
       publicId: createPublicId(),
       content,
       creatorId,
     });
+
+    if (parentPublicId !== undefined) {
+      const parentNote = await this.getNoteByPublicId(parentPublicId);
+
+      if (parentNote === null) {
+        throw new DomainError(`Note with id ${parentPublicId} was not found`);
+      }
+
+      await this.noteRelationsRepository.addNoteRelation(note.id, parentNote.id);
```

Comment:
The strange thing here is a working with relation.

You should either decide if it should be a separate service (NoteRelationsService) or be a part of NoteService.

If it is a separate service, so you should not access its repository from another domain. Access the `NoteRelationsService` instead using [Shared](https://github.com/codex-team/notes.api/tree/main/src/domain/service/shared) service

If it is part of Note Service, it can be left here. Note Service can access `noteRelationsRepository` as its own. But in this case `getParentNoteIdByNoteId` should also be a part of Note Service.